lib/express/spcp.js

```diff
@@ -38,6 +38,9 @@ function config (app, { showLoginPage, serviceProvider, idpConfig, cryptoConfig
             if (assertions.myinfo.personas[id]) {
               id += ' [MyInfo]'
             }
+            if (idp === 'corpPass') {
+              id = `${id.NRIC} / UEN: ${id.UEN}`
+            }
```

Comment:
Not your fault, but the biz logic in lines 38-43 is pretty brittle here. I regret generating the identifier by mangling the function argument. Could we put in perhaps something like.....
```js
const generateIdFrom = idp === 'corpPass'
  ? rawId => `${rawId.NRIC} / UEN: ${rawId.UEN}`
  : rawId => assertions.myinfo.personas[rawId] ? rawId + ' [MyInfo]' : rawId
// ...
.map((rawId, index) => {
  // ...
  const id = generateIdFrom(rawId)
  return { id, assertURL }
})
```

Reply:
Updated.